### PR TITLE
Reduce the number of unexpected warnings during testing

### PR DESCRIPTION
**Pull request by pseudo-rnd-thoughts**
This PR aims to limit the number of unexpected warnings raised during testing to allow spotting of unexpecting errors / warnings

A majority of the changes are to add `with warnings.catch_warnings(record=True)` or `pytest.warns()` about the code that produces the warning

### CODE REVIEW

File: tests/envs/test_make.py [hunk 1]
```diff
@@ -52,13 +53,14 @@ def test_make():
...
 
 def test_make_deprecated():
-    with pytest.raises(
-        gym.error.Error,
-        match=re.escape(
-            "Environment version v0 for `Humanoid` is deprecated. Please use `Humanoid-v4` instead."
-        ),
-    ):
-        gym.make("Humanoid-v0", disable_env_checker=True)
+    with warnings.catch_warnings(record=True):
```
Comment (Markus28) [inline tests/envs/test_make.py:56]: Don't we want to check whether the caught warnings can be safely ignored? As we do in /tests/envs/test_envs.py?

File: tests/envs/test_make.py [hunk 2]
```diff
@@ -228,13 +230,16 @@ def test_make_render_mode():
...
 
     # Make sure that an additional error is thrown a user tries to use the wrapper on an environment with old API
-    with pytest.raises(
-        gym.error.Error,
-        match=re.escape(
-            "You passed render_mode='human' although test/NoHumanOldAPI-v0 doesn't implement human-rendering natively."
-        ),
-    ):
-        gym.make("test/NoHumanOldAPI-v0", render_mode="human", disable_env_checker=True)
+    with warnings.catch_warnings(record=True):
```
Comment (Markus28) [inline tests/envs/test_make.py:233]: Again, do we want to check the warnings?

File: tests/spaces/test_box.py [hunk 3]
```diff
@@ -57,9 +58,11 @@ def test_box_shape_inference(box, expected_shape):
 )
 def test_box_values(value, valid):
     if valid:
-        with pytest.warns(None) as warnings:
+        with warnings.catch_warnings(record=True) as caught_warnings:
             Box(low=value, high=value)
-        assert len(warnings.list) == 0, tuple(warning.message for warning in warnings)
+        assert len(caught_warnings) == 0, tuple(
+            warning.message for warning in caught_warnings
+        )
```
Comment (Markus28) [inline tests/spaces/test_box.py:65]: How about replacing this with
```python
with warnings.catch_warnings():
    warnings.simplefilter("error")
    Box(low=value, high=value)
```
Cf https://docs.pytest.org/en/7.0.x/how-to/capture-warnings.html#additional-use-cases-of-warnings-in-tests

However, I believe that this should only be done in cases in which we do *not* also use `pytest.raises`, because otherwise `pytest.raises` might be satisfied with a warning that was promoted to an exception.

File: tests/utils/test_env_checker.py [hunk 4]
```diff
@@ -47,10 +47,10 @@
 )
 def test_no_error_warnings(env):
     """A full version of this test with all gym envs is run in tests/envs/test_envs.py."""
-    with pytest.warns(None) as warnings:
+    with warnings.catch_warnings(record=True) as caught_warnings:
         check_env(env)
 
-    assert len(warnings) == 0, [warning.message for warning in warnings]
+    assert len(caught_warnings) == 0, [warning.message for warning in caught_warnings]
```
Comment (Markus28) [inline tests/utils/test_env_checker.py:53]: Again, consider replacing it with 

```python
with warnings.catch_warnings():
    warnings.simplefilter("error")
    check_env(env)
```

File: tests/wrappers/test_pixel_observation.py [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
 
 class FakeEnvironment(gym.Env):
-    def __init__(self, render_mode=None):
+    def __init__(self, render_mode="single_rgb_array"):
```
Comment (Markus28) [inline tests/wrappers/test_pixel_observation.py:13]: Why was this line changed?

**Review by Markus28 — APPROVED** [review]
Generally LGTM. CI needs to be fixed but this is probably not related to the PR.
I left some minor comments.

File: tests/envs/test_make.py (hunk 2 above)
Comment (pseudo-rnd-thoughts) [reply tests/envs/test_make.py:233]: repeat of the same general issue

File: tests/spaces/test_box.py (hunk 3 above)
Comment (pseudo-rnd-thoughts) [reply tests/spaces/test_box.py:65]: As we are adding this to a limited number of cases then I dont think this should matter

File: tests/utils/test_env_checker.py (hunk 4 above)
Comment (pseudo-rnd-thoughts) [reply tests/utils/test_env_checker.py:53]: Personally, I like keeping warnings, warnings and errors, errors but this is possible solution
